hpx/hpx_init_impl.hpp

```diff
@@ -208,22 +208,24 @@ namespace hpx
         return init(f, desc_commandline, argc, argv, empty, empty, mode);
     }
 
+    /// \cond NOINTERNAL
+    namespace detail
+    {
+        HPX_EXPORT int init_helper(
+            boost::program_options::variables_map& /*vm*/,
+            util::function_nonser<int(int, char**)> const& f);
+    }
+    /// \endcond
+
     /// \brief Main entry point for launching the HPX runtime system.
     ///
     /// This is a simplified main entry point, which can be used to set up the
     /// runtime for an HPX application (the runtime system will be set up in
     /// console mode or worker mode depending on the command line settings).
-    namespace detail
-    {
-        inline int init_helper(boost::program_options::variables_map& /*vm*/,
-            util::function_nonser<int(int, char**)> const& f, int argc, char** argv)
-        {
-            return f(argc, argv);
-        }
-    }
-
-    inline int init(util::function_nonser<int(int, char**)> const& f,
-        std::string const& /*app_name*/, int argc, char** argv, hpx::runtime_mode mode)
+    inline int
+    init(util::function_nonser<int(int, char**)> const& f,
+        std::string const& /*app_name*/, int argc, char** argv,
```

Comment:
Good point. It was of no use for the old code but can now be utilized. I'll add that. OTOH, removing it wouldn't do any harm either...